android/java/org/chromium/chrome/browser/app/helpers/ImageLoader.java

```diff
@@ -0,0 +1,417 @@
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Locale;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class ImageLoader {
+    private static final List<String> ANIMATED_LIST = Arrays.asList(".gif");
+    private static final String UNUSED_CLIENT_NAME = "unused";
+    private static final String BASE64_ENCODING_PATTERN =
+            "^([A-Za-z0-9+/]{4})*([A-Za-z0-9+/]{4}|[A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{2}==)$";
+    private static final String ATT_VALUE_1000_PX = "1000px";
+    private static final String SVG_TAG = "svg";
+    private static final String DATA_IMAGE_SVG_UTF8_PREFIX = "data:image/svg+xml;utf8,";
+
+    /**
+     * Downloads an image from a given URL, including support for GIF and SVG image types.
+     * @param url URL of the image to download.
+     * @param context Android context used by Glide for applying transformations.
+     * @param isCircular When {@code true}, a circular transformation will be applied.
+     * @param imageView ImageView where the downloaded image will be set.
+     * @param callback Callback used to notify if the image has been set correctly. It can be {@code
+     *         null}.
+     */
+    public static void downloadImage(String url, final Context context, final boolean isCircular,
+            final ImageView imageView, final Callback callback) {
+        if (!isValidImgUrl(url)) {
+            if (callback != null) callback.onLoadFailed();
+            return;
+        }
+
+        Profile profile = Utils.getProfile(false);
+        if (isSvg(url)) {
+            final String validUrl;
+            if (URLUtil.isDataUrl(url)) {
+                if (isBase64Encoded(url)) {
+                    String decodedUrl = decodeBase64SvgUrl(url);
+                    validUrl = sanitizeSvg(decodedUrl);
+                } else if (isUtf8(url)) {
+                    validUrl = sanitizeSvg(url);
+                } else {
+                    // Unsupported URL type.
+                    if (callback != null) callback.onLoadFailed();
+                    return;
+                }
+                if (TextUtils.isEmpty(validUrl)) {
+                    // This may happen for invalid, or corrupted URLs.
+                    if (callback != null) callback.onLoadFailed();
+                    return;
+                }
+            } else {
+                validUrl = url;
+            }
+
+            WebContentsFactory webContentsFactory = new WebContentsFactory();
+            WebContents webContents =
+                    webContentsFactory.createWebContentsWithWarmRenderer(profile, true);
+            webContents.downloadImage(new GURL(validUrl), // Url
+                    false, // isFavIcon
+                    WalletConstants.MAX_BITMAP_SIZE_FOR_DOWNLOAD, // maxBitmapSize
```

Comment:
why is a seemingly generic image loader using `WalletConstants.MAX_BITMAP_SIZE_FOR_DOWNLOAD`?